esp-hal-common/src/rsa/mod.rs

```diff
@@ -185,6 +270,23 @@ where
         self.rsa.clear_interrupt();
         Ok(())
     }
+
+    #[cfg(feature = "async")]
+    pub(crate) fn inner(&self) -> &RSA {
+        &self.rsa.rsa
+    }
+
+    #[cfg(feature = "async")]
+    pub async fn async_exponentiation(
+        &mut self,
+        base: &T::InputType,
+        r: &T::InputType,
+        outbuf: &mut T::InputType,
+    ) {
+        self.start_exponentiation(&base, &r);
+        asynch::RsaFuture::new(self.inner());
```

Comment:
You need to call `.await` on the `RsaFuture`, maybe that will fix it?

Reply:
Unfortunately, it didn't have any effect

Reply:
Yes but you need to await anyways, without the await this function will act just like the blocking function. The `.await` allows the program to do  other things until the RSA interrupt fires and this future resolves.

Reply:
Yes, you're right!
Was fixed already, thank you ❤️